### PR TITLE
Python Updates 2022-07-15

### PR DESCRIPTION
**Pull request by mweinelt**
Second python update run for the big changes and some necessary stabilization work.

Hydra project: https://hydra.nixos.org/jobset/nixpkgs/python-updates
Regressions: https://malob.github.io/nix-review-tools-reports/
Previous run: https://github.com/NixOS/nixpkgs/pull/175805

### CODE REVIEW

**Comment by risicle** [comment]
https://github.com/numba/numba/issues/8242#issuecomment-1184960325

> Neither Numba 0.55.2 and the current RC 0.56.0rc1 support NumPy 1.23. NumPy 1.23 support is planned for the next release (0.57).

File: pkgs/development/python-modules/ddt/default.nix [hunk 1]
```diff
@@ -16,16 +16,21 @@ buildPythonPackage rec {
...
+
+  preCheck = ''
+    # pytest can't import one file even with PYTHONPATH set
+    rm test/test_named_data.py
+  '';
+
   meta = with lib; {
     description = "Data-Driven/Decorated Tests, a library to multiply test cases";
     homepage = "https://github.com/txels/ddt";
+    maintainer = with maintainer; [ ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/ddt/default.nix:33]: maintainers

**Comment by risicle** [comment]
Hmm `thrift` appears to be failing on macos, will have to look another time though

File: pkgs/development/interpreters/python/hooks/sphinx-hook.sh [hunk 2]
```diff
@@ -53,5 +53,5 @@ installSphinxPhase() {
     runHook postInstallSphinx
 }
 
-preBuildPhases+=" buildSphinxPhase"
+preDistPhases+=" buildSphinxPhase"
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/python/hooks/sphinx-hook.sh:56]: Should we also rename ``buildSphinxPhase``?

**Comment by jonringer** [comment]
@mweinelt can you please add the link to the regression report to the PR description? :)

**Comment by SomeoneSerge** [comment]
`cudaSupport = true;` run: https://hercules-ci.com/github/SomeoneSerge/nixpkgs-unfree/jobs/1302

CC @NixOS/cuda-maintainers

**Comment by risicle** [comment]
setuptools >= 61.1 and its new patch is what thrift doesn't like I think.

**Comment by risicle** [comment]
https://github.com/risicle/thrift/commit/d34d67cabe343fab609b477dba51f05c71c952af gets us half the way with thrift, will figure out the rest and send it upstream tomorrow.

**Comment by risicle** [comment]
https://github.com/apache/thrift/pull/2635

**Pull request merged**